Packs/Base/Scripts/GetIncidentsByQuery/GetIncidentsByQuery.py

```diff
@@ -79,11 +117,12 @@ def get_incidents(query, time_field, size, from_date):
             from_datetime = parse_relative_time(from_date)
         if from_datetime:
             args['from'] = from_datetime.isoformat()
-    res = demisto.executeCommand("getIncidents", args)
-    if res[0]['Type'] == entryTypes['error']:
-        error_message = str(res[0]['Contents'])
-        return_error("Failed to get incidents by query: %s error: %s" % (query, error_message))
-    incident_list = res[0]['Contents'].get('data') or []
+    incident_list = []  # type: ignore
+    for page in range(0, int(math.ceil(float(query_size) / float(PAGE_SIZE)))):
```

Comment:
Instead of assuming page size, perhaps we can iterate until we reached enough incidents or until we get a non existing stage

`while len(incidet_list)<query_size and incidents (incidents!=[])`